apps/deploy-web/src/components/new-deployment/TemplateList.tsx

```diff
@@ -56,102 +60,119 @@ export const TemplateList: React.FunctionComponent<Props> = ({ onChangeGitProvid
   }, [templates]);
 
   function onSDLBuilderClick(page: NewDeploymentParams["page"] = "new-deployment") {
+    setEditedManifest("");
+    onTemplateSelected(null);
     setSdlEditMode("builder");
     router.push(UrlService.newDeployment({ step: RouteStep.editDeployment, page }));
   }
 
-  function handleBackClick() {
-    if (previousRoute) {
-      router.back();
-    } else {
-      router.push(UrlService.deploymentList());
+  const propagateUploadedSdl = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const selectedFiles = event.target.files ?? [];
+    const hasFileSelected = selectedFiles.length > 0;
+    if (!hasFileSelected) return;
+    const fileUploaded = selectedFiles[0];
+
+    const reader = new FileReader();
+
+    reader.onload = event => {
+      onTemplateSelected({
+        title: "From file",
+        code: "from-file",
+        category: "General",
+        description: "Custom uploaded file",
+        content: event.target?.result as string
+      });
+      setEditedManifest(event.target?.result as string);
+      setSdlEditMode("yaml");
+
+      router.push(UrlService.newDeployment({ step: RouteStep.editDeployment }));
+    };
+
+    reader.readAsText(fileUploaded);
```

Comment:
there is such logic already. I understand if you want to copypaste for now. However ideally there should be a hook we could re-use

Reply:
Yea might as well do it now